### PR TITLE
Issue 576 extra collection ids

### PR DESCRIPTION
**Pull request by bransteitterbr**
For issue: https://github.com/episphere/connect/issues/576

### CODE REVIEW

**Review by Gbarra9 — APPROVED** [review]
Looks good. I like your `findReplacementTubeLabels` function that is referenced and used to display the correct type/color of the tubes.

File: src/shared.js [hunk 1]
```diff
@@ -1074,6 +1074,33 @@ const buildAvailableCollectionsObject = (specimensList, isPartiallyBoxed) => {
...
+ * Build the available collections object. Remove unusable tubes.
+ * Filter tubes (available collections vs strays) for use in the shipping dashboard.
+ * @param {array<object>} specimensList - list of specimens from Firestore.
+ * @returns {object} { availableCollections, specimensList } - available collections object and the updated specimens list.
+ * Note: Mouthwash tubes are always solo. They belong in available collections. The tube number is always '0007', the bag number is always '0009'.
+ */
+export const findReplacementTubeLabels = (specimensList) => {
+    if (!specimensList || specimensList.length === 0) return { availableCollections: {}, specimensList: [] };
+    const replaceMentTubeLabels = {};
+    const replaceMentLabelRegExp = new RegExp('005[0-4]$');
```
[the remark this answers is not in the document]
Comment (bransteitterbr) [reply src/shared.js]: You are right, I have updated it

File: src/shared.js [hunk 2]
```diff
@@ -1074,6 +1074,33 @@ const buildAvailableCollectionsObject = (specimensList, isPartiallyBoxed) => {
...
+    for (let specimen of specimensList) {
+        const collectionId = specimen[conceptIds.collection.id];
+        if (!collectionId) return;
```
[the remark this answers is not in the document]
Comment (bransteitterbr) [reply src/shared.js]: Also correct it should have been a continue.  I have updated it.

File: src/pages/receipts/csvFileReceipt.js [hunk 3]
```diff
@@ -109,9 +109,11 @@ const confirmFileSelection = () => {
       document.getElementById('modalShowMoreData').querySelector('#closeModal').click(); // closes modal
       showAnimation();
       const response = await getAllBoxes(`bptlPackagesInTransit`);
+      const specimens = await getSpecimensInBoxes(response.data);
```
[the remark this answers is not in the document]
Comment (bransteitterbr) [reply src/pages/receipts/csvFileReceipt.js]: I added a try catch and a simple error modal for this block

**Review by jhflorey — APPROVED** [review]

**Pull request merged**